driver/src/test/java/org/neo4j/driver/internal/async/pool/NettyChannelTrackerTest.java

```diff
@@ -110,6 +110,48 @@ public void shouldDecrementCountForAddress()
         assertEquals( 3, tracker.idleChannelCount( address ) );
     }
 
+    @Test
+    public void shouldDecreaseIdleWhenClosedOutsidePool() throws Throwable
+    {
+        // Given
+        Channel channel = newChannel();
+        tracker.channelCreated( channel, null );
+        assertEquals( 1, tracker.inUseChannelCount( address ) );
+        assertEquals( 0, tracker.idleChannelCount( address ) );
+
+        // When closed before session.close
+        channel.close();
```

Comment:
Maybe `channel.close().sync()`?
`Channel#close()` is async and returns a future.